src/reducers/shelf/spec.ts

```diff
@@ -92,6 +95,67 @@ export function shelfSpecReducer(shelfSpec: Readonly<ShelfUnitSpec> = DEFAULT_SH
       });
     }
 
+    case SHELF_FUNCTION_ADD_WILDCARD: {
+      const {shelfId, fn} = action.payload;
+      return modifyEncoding(shelfSpec, shelfId, (fieldDef: Readonly<ShelfFieldDef | ShelfAnyEncodingDef>) => {
+        const {fn: _fn, ...fieldDefWithoutFn} = fieldDef;
```

Comment:
`_fn` is by convention the _fn that we do not use. 
Here we actually use it, so this is a bit misleading.

`fn: oldFn` ?